File: Java/Euler7/Euler7.java
/*
By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13
 we can see that the 6th prime is 13.
What is the 10 001st prime number?
*/
import java.util.ArrayList;

public class Euler7
{ 
   static ArrayList<Integer> primes = new ArrayList<Integer>();

   public static void main (String []args)
   {
      long startTime = System.nanoTime();//start an execution timer
      primes.add(2);
      for (int i = 3;;i =i+2)
      {
         primes.add(i);
         for(int a = 3;a*a<=i;a++)
         {
            if(i%a==0)
            {
               primes.remove((Integer)i);
            }
         }          
         if(primes.size()==10001)
         {
            System.out.println(primes.get(10000));
            long endTime = System.nanoTime();
            long duration = endTime - startTime;
            System.out.println("The method took: "+(double)duration / 1000000000.0+" seconds");
            break;
         }
      }
   }
}

            

File: Euler7/euler7.swift
/*
 
 By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we can see that the 6th prime is 13.
 
 What is the 10 001st prime number?
 
 */
import Foundation


var nthPrime = 1 // 2 is prime
var number = 1 // 1+2 = 3 is the first unequal number that is potentially a prime

while(nthPrime < 10001){

    // only odd numbers can be primes
    number = number + 2
    // we asume it is a prime
    nthPrime = nthPrime + 1
    
    for divider in 3.stride(to: number, by:2){
        if(number%divider == 0){
            // if it can be divided, its not a prime anyway and we break
            nthPrime = nthPrime - 1
            break
        }
    }

}
print("\(nthPrime)th prime is \(number)")
